arrow-docs/docs/quickstart/generators/README.md

```diff
@@ -0,0 +1,208 @@
+---
+layout: docs-incubator
+title: Writing good generators
+permalink: /check/writing-good-generators
+---
+
+# Writing good generators
+
+This guide aims to expose some common gotchas while writing generators and how to avoid them.
+
+Table of contents:
+- Avoiding monadic composition
+- Avoid manually discarding values when filtering
+- Use `forAllT` when dealing with a large recursive generator
+- Use the inbuilt recursive function for recursive generators
+
+## Avoiding monadic composition
+
+Suppose we want to generate two **independent** values, like two numbers:
+```kotlin:ank:silent
+import arrow.check.gen.Gen
+import arrow.check.gen.monadGen
+import arrow.check.gen.printTreeWith
+import arrow.check.property.Size
+import arrow.check.gen.RandSeed
+
+//sampleStart
+val gen = Gen.monadGen {
+  fx {
+    val a = int(0..5).bind()
+    val b = int(0..5).bind()
+    Pair(a, b)
+  }
+}
+gen.printTreeWith(Size(30), RandSeed(0))
+//sampleEnd
+```
+```
+(2, 2)
+|-> (0, 2)
+|   |-> (0, 0)
+|   --> (0, 1)
+|       --> (0, 0)
+|-> (1, 2)
+|   |-> (0, 2)
+|   |   |-> (0, 0)
+|   |   --> (0, 1)
+|   |       --> (0, 0)
+|   |-> (1, 0)
+|   --> (1, 1)
+|       --> (1, 0)
+|-> (2, 0)
+--> (2, 1)
+    --> (2, 0)
+```
```

Comment:
Hi @1Jajen1 :raised_hands: Just a comment about this console output. Website provides the execution for the code snippets with `kotlin:ank:playground`. Thanks!!

Reply:
I think that modifier is missing in Ank documentation... :open_mouth:

Reply:
I've used that quite a bit in arrow documentation, just forgot that this was available here as well. Thanks for the reminder :)

That does mean I need to tweak the second generators seed to produce (2,2) as well so that it won't time out on the amount of output.

Even (2,4) produces such a large shrink tree that it will likely time out. (Which is good because it is supposed to have a large shrink tree)

Reply:
Thanks @1Jajen1 !! :raised_hands: